io/disk/scsi_add_remove.py

```diff
@@ -33,44 +37,90 @@ def setUp(self):
+            devices.append(line.split('/')[-1].strip(' '))
+        if device in devices:
+            return True
+        else:
+            return False
+
+    def get_scsi_id(self, path):
+        '''
+        calculate and return the scsi_id of a disk
+        '''
+        cmd = "lsscsi"
+        out = process.run(cmd)
+        for lines in out.stdout.splitlines():
+            if path in lines:
+                scsi_num = lines.split()[0].strip("[").strip("]")
+                self.log.info("scsi_num=%s", scsi_num)
+                scsi_num = scsi_num.replace(":", " ")
+                return scsi_num
 
     def test(self):
         '''
         Function where test is executed
         '''
-        device_list = []
-        cmd = "ls -l /dev/disk/by-path/"
-        output = process.run(cmd)
-        for lines in output.stdout.splitlines():
-            if self.pci_device in lines:
-                device_list.append(lines.split()[-1])
-        if not device_list:
-            self.log.warning("No devices under the given PCI device")
-        else:
-            for device_id in device_list:
-                device = device_id.split()[-1].strip("../*")
-                self.log.info("device = %s", device)
-                cmd = "lsscsi"
-                output = process.run(cmd)
-                for lines in output.stdout.splitlines():
-                    if device in lines:
-                        scsi_num = lines.split()[0].strip("[").strip("]")
-                        self.log.info("scsi_num=%s", scsi_num)
-                        scsi_num_seperated = scsi_num.replace(":", " ")
-                        self.log.info("Deleting %s", scsi_num)
-                        genio.write_file("/sys/block/%s/device/delete"
-                                         % device, "1")
-                        time.sleep(5)
-                        self.log.info("%s deleted", scsi_num)
-                        self.log.info("adding back %s", scsi_num)
-                        process.run("echo scsi add-single-device %s > \
-                                     /proc/scsi/scsi", scsi_num_seperated)
-                        time.sleep(5)
-                        self.log.info("%s Added back", scsi_num)
+        self.log.info("device lists : %s " % self.device_list)
+        for device in self.device_list:
+            scsi_id = self.get_scsi_id(device)
+            process.run("\nlsscsi\n")
+            self.log.info("\nDeleting %s = %s\n" % (device, scsi_id))
+            genio.write_file("/sys/block/%s/device/delete" % device, "1")
+            time.sleep(5)
+            if self.is_exists_scsi_device(device) is True:
+                self.err_paths.append(device)
+            else:
+                self.log.info("\n%s = %s deleted\n" % (device, scsi_id))
+            self.log.info("\nadding back %s = %s\n" % (device, scsi_id))
+            part = "scsi add-single-device %s" % scsi_id
+            genio.write_file("/proc/scsi/scsi", part)
```

Comment:
Is this same as `process.run("echo scsi add-single-device %s > /proc/scsi/scsi", scsi_id)` ?

Reply:
Just asking for confirmation.

Reply:
yes